src/prog_algs/predictors/toe_prediction_profile.py

```diff
@@ -84,3 +86,47 @@ def prognostic_horizon(self, criteria_eqn, ground_truth, **kwargs) -> Dict[str,
         from ..metrics import prognostic_horizon
         return prognostic_horizon(self, criteria_eqn, ground_truth, **kwargs)
 
+    def plot(self, ground_truth : dict = None , alpha : float = None, print : bool = True) -> dict: # use ground truth, alpha if given,
+        """Produce an alpha-beta plot depicting the TtE distribution by time of prediction.
+
+        Args:
+            ground_truth : dict = None
+                Optional dictionary argument containing event and its respective ground truth value; none by default and plotted if specified
+            alpha : float
+                Optional alpha value; none by default and plotted if specified
+            print : bool = True
+                Optional bool value; specify whether to display generated plots or not
+        Returns:
+            dict
+                Collection of generated matplotlib figures for each event in profile
+        """
+        result_figs = {}
+        for t,v in self.items():
+            for key in v.keys():
+                if key not in result_figs:
+                    # Prepare Figure for Plot
+                    fig_window = plt.figure() # Create new figure for this event key
+                    fig_sub = fig_window.subplots()
+                    fig_sub.grid()
+                    fig_sub.set_title(f"Event {key} Plot")
+                    fig_sub.set_xlabel('Time of Prediction (s)') # time to prediction
+                    fig_sub.set_ylabel('Time to Event (s)') # time to event
+                    result_figs[key] = fig_window
+                # Create scatter plot for this event
+                samples = v.sample(100) # sample distribution (red scatter plot)
```

Comment:
v.sample can be moved to before the for key in v.keys() loop - that way you don't have to resample for each key.